src/dev/stage_by_owner.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import simpleGit from 'simple-git';
+import fs from 'fs';
+import path from 'path';
+
+import { run } from '@kbn/dev-cli-runner';
+import { getOwningTeamsForPath, getCodeOwnersEntries } from '@kbn/code-owners';
+import { asyncForEach } from '@kbn/std';
+import { inspect } from 'util';
+
+const git = simpleGit();
+
+const CODEOWNERS_PATH = path.join('.github', 'CODEOWNERS');
+
+interface File {
+  path: string;
+  staged: boolean;
+}
+
+// Function to get the list of changed files
+const getChangedFiles = async (): Promise<File[]> => {
+  const { staged, files } = await git.status();
+  return files.map((file) => ({ path: file.path, staged: staged.includes(file.path) }));
+};
+
+run(
+  async ({ flags, log }) => {
+    const {
+      _: [owner],
+    } = flags;
+
+    if (!fs.existsSync(CODEOWNERS_PATH)) {
+      log.error('CODEOWNERS file not found!');
+      process.exit(1);
+    }
```

Comment:
`getCodeOwnersEntries` will throw an `Error` if there's no `CODEOWNERS` file. 

You could remove this along with L11-12 and L21, catch the exception on L51 then log the error there instead.